lib/SILOptimizer/Analysis/MemoryBehavior.cpp

```diff
@@ -241,10 +241,17 @@ MemBehavior MemoryBehaviorVisitor::visitApplyInst(ApplyInst *AI) {
   MemBehavior Behavior = MemBehavior::None;
 
   // We can ignore mayTrap().
-  if (ApplyEffects.mayReadRC() ||
-      (InspectionMode == RetainObserveKind::ObserveRetains &&
-       ApplyEffects.mayAllocObjects())) {
-    Behavior = MemBehavior::MayHaveSideEffects;
+  if (any_of(enumerate(AI->getArgumentOperands()),
```

Comment:
Hm, I don't see anything obviously wrong here. Can you run the compiler in the debugger and dump the function and the apply instruction and op.Index?